Lib/inspect.py

```diff
@@ -1805,6 +1806,9 @@ def _shadowed_dict(klass):
                 return class_dict
     return _sentinel
 
+def _shadowed_dict(klass):
+    return _shadowed_dict_from_mro_tuple(_static_getmro(klass))
```

Comment:
I considered inlining it, but I thought this made it more readable. It also preserves `git blame` better, and having to pass through this function doesn't seem to have much impact on performance